platform/ios/scripts/script_resources/MapboxDemo/MapboxDemo/ViewController.swift

```diff
@@ -0,0 +1,239 @@
+        setupAndAddMapView()
+        setupGestureHandling()
+    }
+    
+    // MARK: - Map object interaction
+    
+    @objc func didTapMap(sender: UITapGestureRecognizer) {
+        
+        // Search for annotation at tap point
+        // If the annotation is found at the tap point then modify some of its properties
+        // If the annotation is not found at the tap point, reset the annotation back to the way it was when originally added
+        if let layer = mapView.style?.layer(withIdentifier: annotationIdentifier) as? MGLFillExtrusionStyleLayer {
+            let point = sender.location(in: mapView)
+            let features = mapView.visibleFeatures(at: point, styleLayerIdentifiers: [annotationIdentifier])
+            if features.count > 0 {
+                layer.fillExtrusionOpacity = MGLStyleValue(rawValue: 0.9)
+                layer.fillExtrusionHeight = MGLStyleValue(rawValue: 5000)
+                let camera = MGLMapCamera(lookingAtCenter: centerCoordinate, fromDistance: 100000, pitch: 60, heading: 180)
+                mapView.fly(to: camera, withDuration: 8, completionHandler: nil)
+                calloutView.alpha = 1.0
+            } else {
+                layer.fillExtrusionOpacity = MGLStyleValue(rawValue: 0.5)
+                layer.fillExtrusionHeight = MGLStyleValue(rawValue: 0)
+                let camera = MGLMapCamera(lookingAtCenter: centerCoordinate, fromDistance: 100000, pitch: 60, heading: 0)
+                mapView.fly(to: camera, withDuration: 1, completionHandler: nil)
+                calloutView.alpha = 0
+            }
+        }
+    }
+    
+    // MARK: - Initialization and setup
+    
+    /*
+     Simple (but somewhat styled) stub of a view to present
+     contextual information about tapped objects on the map.
+     */
+    fileprivate func setupCalloutView() {
+        calloutView.alpha = 0
+        let blurView = UIVisualEffectView(effect: UIBlurEffect(style: .light))
+        blurView.frame = calloutView.bounds
+        blurView.layer.cornerRadius = 5
+        blurView.clipsToBounds = true
+        blurView.translatesAutoresizingMaskIntoConstraints = false
+        calloutView.insertSubview(blurView, at: 0)
+        blurView.topAnchor.constraint(equalTo: calloutView.topAnchor).isActive = true
+        blurView.leftAnchor.constraint(equalTo: calloutView.leftAnchor).isActive = true
+        blurView.rightAnchor.constraint(equalTo: calloutView.rightAnchor).isActive = true
+        blurView.bottomAnchor.constraint(equalTo: calloutView.bottomAnchor).isActive = true
+    }
+    
+    /*
+     Create and add a Mapbox map view to the the view
+     intended to contain it. Setup with the style, initial
+     location in the world, and make this object the delegate
+     of the map. Finally, set up a camera to control how the
+     user will view the map initially.
+     */
+    fileprivate func setupAndAddMapView() {
+        mapView = MGLMapView(frame: mapViewContainer.bounds)
```

Comment:
Can you explain your thought process for doing the map view setup in code, rather than entirely in the storyboard?

Reply:
Just because of https://github.com/mapbox/mapbox-gl-native/issues/10072. I think we should avoid having any warnings (no mater how cosmetic) in this simple example.